src/core/Akka.Cluster/Serialization/ClusterMessageSerializer.cs

```diff
@@ -141,38 +155,37 @@ public override object FromBinary(byte[] bytes, string manifest)
         // </MsgRead>
 
         public override string Manifest(object o)
-        {
-            switch (o)
-            {
-                case InternalClusterAction.Join _:
-                    return JoinManifest;
-                case InternalClusterAction.Welcome _:
-                    return WelcomeManifest;
-                case ClusterUserAction.Leave _:
-                    return LeaveManifest;
-                case ClusterUserAction.Down _:
-                    return DownManifest;
-                case InternalClusterAction.InitJoin _:
-                    return InitJoinManifest;
-                case InternalClusterAction.InitJoinAck _:
-                    return InitJoinAckManifest;
-                case InternalClusterAction.InitJoinNack _:
-                    return InitJoinNackManifest;
-                case ClusterHeartbeatSender.Heartbeat _:
-                    return HeartBeatManifestPre1419;
-                case ClusterHeartbeatSender.HeartbeatRsp _:
-                    return HeartBeatRspManifestPre1419;
-                case InternalClusterAction.ExitingConfirmed _:
-                    return ExitingConfirmedManifest;
-                case GossipStatus _:
-                    return GossipStatusManifest;
-                case GossipEnvelope _:
-                    return GossipEnvelopeManifest;
-                case ClusterRouterPool _:
-                    return ClusterRouterPoolManifest;
-                default:
-                    throw new ArgumentException($"Can't serialize object of type [{o.GetType()}] in [{GetType()}]");
-            }
+            => Manifest(o.GetType());
+        
+        public override string Manifest(Type type)
```

Comment:
I'd prefer it if we didn't add this

Reply:
Done, reverted, changes moved to the spec instead